### DISCUSSION TITLE
Re-use Addresses for Invoices to decrease fees associated with multiple Inputs

### DISCUSSION THREAD

**Discussion by cosifresco** [btcpayserver/btcpayserver#5956] [Ideas / Feature Requests]
We really need to start reusing addresses for invoices. Bitcoin fees for sending transactions out of the BTCPay server wallet are becoming criminal if multiple inputs are involved.

I'm proposing an option that can be enabled/disabled for re-using addresses to minimize the inputs and save on miner fees when eventually sending Bitcoin out of BTCPay server. 

Possible Solution: 

1) Address re-use based on invoices that are active and not expired.
2) If an invoice is active and someone makes another invoice that invoice will have a new address. 
3) Once an invoice is expired/settled the address will be thrown back into the queue of available addresses for new invoices.
4) If an address is thrown into a status that is not expired/settled it will not be thrown back into the queue until the invoice is reviewed and marked as ready for re-use

**Comment by dennisreimann** [comment]
> Bitcoin fees for sending transactions out of the BTCPay server wallet are becoming criminal if multiple inputs are involved.

Re-using addresses does not change that, because addresses aren't UTXOs. Keep in mind that even if you receive twice to the same address, you will have two separate UTXOs and hence two inputs for a future consolidating transaction.

**Reply by cosifresco** [reply]
You're right, I always forget about this
